integration-docs: Update for new format (Mention, Netlify, Opbeat, Open Collective)

PieterCK opened the pull request:
Apply integration-docs' new format.

Fixes part of https://github.com/zulip/zulip/issues/29592.

Updated docs:

<details>
<summary>Mention</summary>

![image](https://github.com/user-attachments/assets/77e97d76-b866-4883-82d2-67430a5c7e14)

</details>

<details>
<summary>Netlify</summary>

![image](https://github.com/user-attachments/assets/25f022ae-e642-4ae8-8ad3-afcbc8903cfc)

</details>

<details>
<summary>Open Collective</summary>

![image](https://github.com/user-attachments/assets/b137e9b7-4b00-4a19-99f9-049a86eed8fc)

</details>

<details>

<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

PieterCK commented:
@zulipbot add "buddy review"

@adnan-td @Niloth-p guys, please check this one out! Thanks

adnan-td commented on zerver/webhooks/opencollective/doc.md:
```diff
@@ -1,19 +1,25 @@
 
 ![](/static/images/integrations/opencollective/001.png)
 
-In the future, this integration can be developed in order to
-support other types of *Activity* such as *New Transaction*, *Subscription Canceled* etc.
```
Can this be added as a note?

Niloth-p replied:
I agree with Pieter that we can remove this from this doc, as the docs are aimed at users. But to retain the point, we could add it to some other platform where follow-ups are tracked?

laurynmm replied:
Removing this from the documentation seems fine.

adnan-td commented:
LGTM!

Niloth-p commented on zerver/webhooks/mention/doc.md:
```diff
@@ -1,27 +1,38 @@
-1. {!generate-integration-url.md!}
+1. {!generate-webhook-url-basic.md!}
 
 1. Go to your Mention feed, and click on your profile in the top-right
```
Replace fullstops with commas?

Niloth-p commented on zerver/webhooks/mention/doc.md:
```diff
@@ -1,27 +1,38 @@
 1. Click **Explore Mention on Zapier**. Search for "webhooks" in
-   the search bar, and click on **Webhooks by Zapier**. Scroll down
-   and look for **Add Webhook posts for new Mentions**, and click on
+   the search bar, and click on **Webhooks by Zapier**. Scroll down,
```
I think we're supposed to remove phrases like "Scroll down" as per the guidelines.

Niloth-p commented on zerver/webhooks/opencollective/doc.md:
```diff
@@ -1,19 +1,25 @@
-when a new member signs-up on an **Open Collective** page.
+# Zulip Open Collective integration
+
+Get Zulip notifications for new sign-ups on your **Open Collective** page.
```
This reads much better, good one!

Niloth-p commented on zerver/webhooks/mention/doc.md:
```diff
@@ -1,27 +1,38 @@
-   by Zapier POST**. When prompted, set **URL** to the URL constructed
-   above, and set **Payload Type** to **JSON**. After **Test this Step**,
-   click **Finish**.
+1. Follow the on-screen steps to link your Mention account to Zapier,
```
Although this is not particularly necessary, it could read better if we format the sequencing of these (6th & 7th in this PR) instructions in a similar manner, since they have some common aspects.

Niloth-p reviewed (commented):
Lgtm, Pieter!
Just suggested a few readability improvements from other PRs.

PieterCK commented:
Thanks for the review guys, I mostly only update the Mention doc to address the reviews.

@zulipbot remove "buddy review"

PieterCK commented:
@zulipbot add "mentor review"

@sbansal1999 Please review this too. Thanks

laurynmm commented on zerver/webhooks/mention/doc.md:
```diff
@@ -1,27 +1,38 @@
 1. Go to your Mention feed, and click on your profile in the top-right
-   corner. Select **Settings**. Click on **Integrations**. Select the
-   **Zapier** integration.
+   corner. Select **Settings**, click on **Integrations**, and Select
```
The second "select" shouldn't be capitalized here.

laurynmm replied:
Looks like this is still pending an update.

laurynmm commented on zerver/webhooks/mention/doc.md:
```diff
@@ -1,27 +1,38 @@
-   and look for **Add Webhook posts for new Mentions**, and click on
+   the search bar, and click on **Webhooks by Zapier**. Look for
+   **Add Webhook posts for new Mentions**, and click on
    **Use this Zap**. Click on **Create this Zap**.
```
Let's revise to "Click **Create this Zap**."

laurynmm commented on zerver/webhooks/mention/doc.md:
```diff
@@ -1,27 +1,38 @@
-   above, and set **Payload Type** to **JSON**. After **Test this Step**,
-   click **Finish**.
+1. Follow the on-screen steps to link your Mention account to Zapier,
+   select your Mention **Account ID**, and **Alert** when prompted.
```
Let's make this a separate sentence: "Select your Mention **Account ID** and **Alert** when prompted."

laurynmm commented on zerver/webhooks/opencollective/doc.md:
```diff
@@ -1,19 +1,25 @@
-your desired collective page, then go to *Settings* -> *Webhooks*, paste the
-bot URL and choose *Activity* -> *New Member*.
+your desired collective page, then go to **Settings** > **Webhooks**, paste the
+generated URL, and choose **Activity** > **New Member**.
```
The current instructions make it seem like you can set up this notification for any collective on the site, but really you can only create it for a collective you're an admin for. Also we should  separate out the navigation and configuration  steps.

```
1. In Open Collective, navigate to one of the collectives under
   **My Collectives**. Go to **Settings** > **Webhooks**.

1. Set **URL** to the URL generated above, and choose **Activity** >
   **New Member**.
```

laurynmm replied:
Seems to not have been updated for the comment above.

laurynmm reviewed (commented):
@PieterCK - Thanks for working on these documentation updates. I made a few comments; let me know if you have any questions.

More generally:
- It looks like Opbeat no longer exists, so we should likely just remove that integration completely.
- I think you can go ahead and update `zerver/lib/integrations.py` for Open Collective to be in the "financial" category. That can be in the same commit with the documentation updates.

PieterCK commented:
Update: addressed reviews, deleted Opbeat and updated the screenshots in PR description

@laurynmm Thanks for reviewing. let me know whether or not the changes looks good to you

laurynmm commented:
@PieterCK - Looks like two of the review comments are still pending updates. Ping me again once those are addressed for another review.

PieterCK commented:
@laurynmm - Ah, my bad. Sorry about that. I’ve updated the PR description and addressed the remaining comments.

laurynmm commented:
@PieterCK - No worries and thanks for the quick update. Looks good and marking for integration review!

Noting that Netlify needs a logo update and the example message content in Mention should be improved; posting that in [this CZO conversation](https://chat.zulip.org/#narrow/stream/19-documentation/topic/integration.20docs.20that.20need.20example.20message.2Flogo.20updates/near/1920281).

timabbott commented:
Merged, thanks @PieterCK and reviewers!
